nixos/modules/services/monitoring/scrutiny.nix

```
@@ -0,0 +1,216 @@
+          type = lib.types.bool;
+          default = false;
+          description = lib.mdDoc "Skip TLS verification when connecting to InfluxDB.";
+        };
+
+        token = lib.mkOption {
+          type = lib.types.str;
+          default = "";
+          description = lib.mdDoc "Authentication token for connecting to InfluxDB.";
+        };
+
+        org = lib.mkOption {
+          type = lib.types.str;
+          default = "";
+          description = lib.mdDoc "InfluxDB organisation under which to store data.";
+        };
+
+        bucket = lib.mkOption {
+          type = lib.types.str;
+          default = "";
+          description = lib.mdDoc "InfluxDB bucket in which to store data.";
+        };
+      };
+
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Open the default ports in the firewall for Scrutiny.";
+      };
+
+      logLevel = lib.mkOption {
+        type = lib.types.enum [ "INFO" "DEBUG" ];
+        default = "INFO";
+        description = lib.mdDoc "Log level for Scrutiny.";
+      };
+
+      collector = {
+        enable = lib.mkEnableOption "Enables the scrutiny metrics collector.";
+
+        package = lib.mkPackageOptionMD pkgs "scrutiny-collector" { };
+
+        endpoint = lib.mkOption {
+          type = lib.types.str;
+          default = "http://localhost:8080";
+          description = lib.mdDoc "Scrutiny app API endpoint for sending metrics to.";
+        };
+
+        interval = lib.mkOption {
+          type = lib.types.int;
+          default = 15;
+          description = lib.mdDoc "Interval (in minutes) on which to collect information about disks.";
+        };
+      };
+    };
+  };
+
+  config = {
+    services.influxdb2 = lib.mkIf cfg.influxdb.enable {
+      inherit (cfg) enable;
+    };
+
+    networking.firewall = lib.mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.port ];
+    };
+
+    services.smartd = lib.mkIf cfg.collector.enable {
+      enable = true;
+      extraOptions = [
+        "-A /var/log/smartd/"
+        "--interval=600"
+      ];
+    };
+
+    systemd = {
+      services = {
+        scrutiny = lib.mkIf cfg.enable {
+          description = "Hard Drive S.M.A.R.T Monitoring, Historical Trends & Real World Failure Thresholds";
+          wantedBy = [ "multi-user.target" ];
+          after = [ "network.target" ];
+          serviceConfig = {
+            DynamicUser = true;
+            ExecStart = "${lib.getExe cfg.package} start --config ${mkScrutinyCfg cfg}";
+            Restart = "always";
+            StateDirectory = "scrutiny";
+            StateDirectoryMode = "0750";
+          };
+        };
+
+        scrutiny-collector = lib.mkIf cfg.collector.enable {
+          description = "Scrutiny Collector Service";
+          environment = {
+            COLLECTOR_API_ENDPOINT = "${cfg.collector.endpoint}";
+          };
+          serviceConfig = {
+            Type = "oneshot";
```

This is a continuously running process, not a program expected to terminate soon-ish right?
```suggestion
            Type = "simple";
```

---

No, this is not long running. It's a one-shot that uses a systemd timer to spawn on an interval. The upstream uses cron for the same purpose.

---

Oh right, i misread!